.travis.yml

```diff
@@ -0,0 +1,15 @@
+sudo: false
+os:
+ - linux
+ - osx
+language: d
+d:
+ - ldc
+ - dmd
+branches:
+  only:
+    - master
+script:
+ - make test unittest_flags="-cov -main -unittest"
+after_success:
+ - bash <(curl -s https://codecov.io/bash)
```

Comment:
can be activated here: https://github.com/integration/codecov

(though it might be that CodeCov has some troubles which your atypical project structure)